### PR TITLE
deprecation(semver): deprecate `ltr()` and `gtr()`

### PR DESCRIPTION
**Pull request by timreichen**
- ~~renames `ltr()` to `lessThanRange()` and deprecates the old function~~
- ~~renames `gtr()` to `greaterThanRange()` and deprecates the old function~~
- deprecates `ltr()`
- deprecates `gtr()`

### CODE REVIEW

File: semver/greater_than_range.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+import type { Range, SemVer, SemVerRange } from "./types.ts";
+import { rangeMax } from "./range_max.ts";
+import { greaterThan } from "./greater_than.ts";
+
+/** Checks to see if the version is greater than all possible versions of the range. */
```
Comment (iuioiua) [inline semver/greater_than_range.ts:6]: If one asks, "Is this version greater than the range?" perhaps it's worth swapping the argument order. Ditto with the other new functions. What do we think?

File: semver/gtr.ts [hunk 2]
```diff
@@ -1,12 +1,11 @@
...
-  version: SemVer,
-  range: SemVerRange | Range,
-): boolean {
-  return gt(version, rangeMax(range));
+/**
+ * Checks to see if the version is greater than all possible versions of the range.
+ * @deprecated (will be removed after 0.215.0) Use {@linkcode greaterThanRange} instead.
+ */
+export function gtr(version: SemVer, range: SemVerRange | Range): boolean {
+  return greaterThanRange(range, version);
```
Comment (iuioiua) [inline semver/gtr.ts:10]: Can you please use the new [`warnOnDeprecatedApi()`](https://github.com/denoland/deno_std/blob/0a32c8f5e82f939e8a46f8c027c1fb5b3040590a/internal/warn_on_deprecated_api.ts#L33)? Ditto with other deprecated functions.

**Review by iuioiua — CHANGES_REQUESTED** [review]
I agree with the changes. Nice.

File: semver/greater_than_range.ts (hunk 1 above)
Comment (timreichen) [reply semver/greater_than_range.ts:6]: My train of thought was to pass the context element first, e.g. the range. But that makes a lot of sense.

File: semver/gtr.ts (hunk 2 above)
Comment (timreichen) [reply semver/gtr.ts:10]: I'll have a look

**Comment by kt3k** [comment]
I wonder what are the use cases of these functions. I guess most users are only interested in `testRange` for comparing a version and a range. In what cases, would people try checking these conditions?

**Comment by timreichen** [comment]
> I wonder what are the use cases of these functions. I guess most users are only interested in `testRange` for comparing a version and a range. In what cases, would people try checking these conditions?

I have no idea. I think it is pretty trivial to get the min/max version of a range and compare it to another version. But I guess there must be a reason why it was ported over.

File: semver/gtr.ts (hunk 2 above)
Comment (timreichen) [reply semver/gtr.ts]: I like that new API, but this seems like the addition of this should be automated to stay in sync with`@deprecated` tags.

Comment (iuioiua) [reply semver/gtr.ts]: I agree, and I think we should explore that idea in a separate issue. But for now, can we please use `warnOnDeprecatedApi()`?

Comment (timreichen) [reply semver/gtr.ts]: done

**Comment by kt3k** [comment]
> But I guess there must be a reason why it was ported over.

I feel we blindly ported everything from `npm:semver` and these wasn't a discussion about utilities of these APIs.

I'd suggest we should just deprecate these without replacement APIs because they don't seem common operation and also the alternative ways to perform these are trivial as you point out. The deprecation note would be something like `use greaterThan(version, rangeMax(range)) instead`

**Comment by timreichen** [comment]
> > But I guess there must be a reason why it was ported over.
> 
> I feel we blindly ported everything from `npm:semver` and these wasn't a discussion about utilities of these APIs.
> 
> I'd suggest we should just deprecate these without replacement APIs because they don't seem common operation and also the alternative ways to perform these are trivial as you point out. The deprecation note would be something like `use greaterThan(version, rangeMax(range)) instead`

done

File: semver/gtr.ts [hunk 3]
```diff
@@ -1,12 +1,22 @@
...
 import type { Range, SemVer, SemVerRange } from "./types.ts";
 import { rangeMax } from "./range_max.ts";
-import { gt } from "./gt.ts";
+import { greaterThan } from "./greater_than.ts";
+import { warnOnDeprecatedApi } from "../internal/warn_on_deprecated_api.ts";
 
-/** Checks to see if the version is greater than all possible versions of the range. */
+/**
+ * Checks to see if the version is greater than all possible versions of the range.
+ * @deprecated (will be removed after 0.215.0) Use `greaterThan(version, rangeMax(range))` instead.
```
Comment (kt3k) [inline semver/gtr.ts:9]:
```suggestion
 * @deprecated (will be removed after 0.217.0) Use `greaterThan(version, rangeMax(range))` instead.
```

File: semver/ltr.ts [hunk 4]
```diff
@@ -1,12 +1,22 @@
...
 import type { Range, SemVer, SemVerRange } from "./types.ts";
-import { lt } from "./lt.ts";
+import { lessThan } from "./less_than.ts";
 import { rangeMin } from "./range_min.ts";
+import { warnOnDeprecatedApi } from "../internal/warn_on_deprecated_api.ts";
 
-/** Less than range comparison */
+/**
+ *  Less than range comparison
+ * @deprecated (will be removed after 0.215.0) Use `lessThan(version, rangeMin(range))` instead.
```
Comment (kt3k) [inline semver/ltr.ts:9]:
```suggestion
 * @deprecated (will be removed after 0.217.0) Use `lessThan(version, rangeMin(range))` instead.
```

**Comment by iuioiua** [comment]
Towards #4228

**Review by iuioiua — APPROVED** [review]
LGTM. I've made some modifications and opened a new issue (#4273) to track the deprecation. Does this new way of deprecation look good to us?

**Comment by kt3k** [comment]
> LGTM. I've made some modifications and opened a new issue (https://github.com/denoland/deno_std/issues/4273) to track the deprecation. Does this new way of deprecation look good to us?

Thanks! #4273 Looks good to me.

**Review by kt3k — APPROVED** [review]
LGTM

**Pull request merged**